### ISSUE TITLE
The Flag review menu items overflow in locales with long strings (de) on small screens

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#10449]
### Describe the problem and steps to reproduce it:
1. Log into AMO and change your locale to `/de/`
2. Navigate to an add-on's All reviews page in mobile view
3. Click on Flag for one of the available reviews

### What happened?
The Flag menu items overflow

### What did you expect to happen?
No layout issues

![image](https://user-images.githubusercontent.com/31961530/115418706-5edb5980-a202-11eb-8993-34ec42736001.png)

**Comment by championshuttler** [comment]
May be for another issue but this looks really bad, i think we can prevent `Flag` to break 

![image](https://user-images.githubusercontent.com/22130317/115432540-1d5ba600-a224-11eb-9d5a-55ac33a7919e.png)

@AlexandraMoga can you confirm if you see this? I am FF desktop and changed the size to mobile screen

**Comment by championshuttler** [comment]
Possible regression from `word-break: break-word;` @willdurand ?

**Comment by bobsilverberg** [comment]
Ugh, we just fixed that awhile ago.

**Comment by bobsilverberg** [comment]
I can confirm this is a regression from `word-break: break-word;`. ☹️

**Comment by championshuttler** [comment]
> I can confirm this is a regression from `word-break: break-word;`. ☹️

May be if we add `word-break: break-word; just for safari globally via media query @bobsilverberg ? and then prevent that rendering here for this class, idk how to do that though, have to look up?

**Comment by bobsilverberg** [comment]
I guess we can specify `word-break: normal;` for this element, and hope this doesn't occur in too many other places? I know this specific element was an issue before.

None of this addresses the original issue filed above though, this is just about the second screenshot added by @championshuttler.

**Comment by bobsilverberg** [comment]
@championshuttler Further to the above, can you file a separate issue for this? It's not the same problem. That is unless you plan to try to fix both issues with a single patch.

**Comment by championshuttler** [comment]
Yeah sure, i just found out this while triaging the mentioned bug

**Comment by bobsilverberg** [comment]
I think that the problem shown in the second screenshot is more critical to fix, so a separate issue probably makes sense.

**Comment by championshuttler** [comment]
Filed #10452

**Comment by willdurand** [comment]
> I can confirm this is a regression from `word-break: break-word;`. ☹️

Ugh.

**Comment by willdurand** [comment]
@bobsilverberg I am thinking about removing the use of `anywhere` for now, I've done that in addons-blog already: https://github.com/mozilla/addons-blog/blob/d623a63e7f97a9b2867f3837ab91a8c226838e11/src/assets/css/styles.scss#L15-L19

**Comment by bobsilverberg** [comment]
> @bobsilverberg I am thinking about removing the use of `anywhere` for now, I've done that in addons-blog already: [mozilla/addons-blog@`d623a63`/src/assets/css/styles.scss#L15-L19](https://github.com/mozilla/addons-blog/blob/d623a63e7f97a9b2867f3837ab91a8c226838e11/src/assets/css/styles.scss#L15-L19)

I suppose you can try, although it seems like that would be undoing everything we did previously to deal with this issue. We started using `anywhere` as a default some time ago so we didn't have to keep overriding individual items with `anywhere` each time we found one, which seemed to happen quite frequently. Do you have a better solution, or are we just going to go back to adding `anywhere` to every individual item that needs it? If the latter, we're probably going to have a whole bunch of things to fix.

**Comment by willdurand** [comment]
> it seems like that would be undoing everything we did previously to deal with this issue.

Yes.

> Do you have a better solution

I am hoping to use `word-break: break-word` everywhere and that it will work. I know we introduced `anywhere` for a reason but I am also hoping to (1) remember why and (2) come up with an alternative.

**Comment by bobsilverberg** [comment]
> > it seems like that would be undoing everything we did previously to deal with this issue.
> 
> Yes.
> 
> > Do you have a better solution
> 
> I am hoping to use `word-break: break-word` everywhere and that it will work. I know we introduced `anywhere` for a reason but I am also hoping to (1) remember why and (2) come up with an alternative.

The "why" is because it was the only thing that would work in certain situations. I can't explain why it was the only thing that would work, but that was the case. No other css property combination seemed to work in those cases. It always had to do with long unbroken strings refusing to wrap without the use of `overflow-wrap: anywhere;`. My concern about reverting that change is that we're not going to remember all of the cases where we needed it, and hence introduce a bunch of `prod_bug`s that will need to be addressed. Maybe if we go back to the commit where we introduced that change we can see the specific elements it was removed from, and that will give us a clue about where it was needed.

As discussed above, the reason we made it a default was because we wanted to avoid having to add it to each of these cases individually. This was something about which there was quite a bit of discussion, which is another reason I'm not sure we want to undo that change. As I understand it, it was working for quite some time, and the current problems are actually the result of adding `word-break: break-word;` to support Safari. I would argue that supporting Safari isn't actually that important, but I guess affecting Safari also means affecting Firefox for iOS, and we probably should avoid a bad experience on that browser.

**Comment by willdurand** [comment]
I guess you're right. AMO on Safari isn't _that_ useful.. so maybe we should simply undo the recent changes to attempt to support and fix Safari instead?

**Comment by willdurand** [comment]
In any case, I think most of the discussion is about something else that is not what QA reported initially in the issue. The "Flag" link has nothing to do with the tooltip that breaks the page.

Without `word-break: break-word;`, it is still happening.

**Comment by AlexandraMoga** [comment]
@willdurand I see that the tooltip overflow has been fixed, but text inside the tooltip is still truncated:

[de]
![image](https://user-images.githubusercontent.com/31961530/117622606-78c1e980-b17b-11eb-9041-12ad33ed86ff.png)

[fr]
![image](https://user-images.githubusercontent.com/31961530/117622677-9000d700-b17b-11eb-832d-87e47be1a3ee.png)

**Comment by willdurand** [comment]
> @willdurand I see that the tooltip overflow has been fixed, but text inside the tooltip is still truncated:
> 
> [de]
> ![image](https://user-images.githubusercontent.com/31961530/117622606-78c1e980-b17b-11eb-9041-12ad33ed86ff.png)
> 
> [fr]
> ![image](https://user-images.githubusercontent.com/31961530/117622677-9000d700-b17b-11eb-832d-87e47be1a3ee.png)

Yes, the text is too long and there is little to nothing we can do...

**Comment by AlexandraMoga** [comment]
Could we at least add an ellipsis?

**Comment by championshuttler** [comment]
Yes I think ellipsis would be better than truncation

**Comment by willdurand** [comment]
Yes, unless we find a way to wrap menu items..

**Comment by AlexandraMoga** [comment]
Ok, I can file a follow up issue to think about this further.
